packages/commerce-sdk-react/src/hooks/ShopperBaskets/query.ts

```diff
@@ -40,8 +41,10 @@ function useBasket(
     const defaultOptions = {
         enabled: !!parameters.basketId
     }
+    const queryKey = queryMap.basket({basketId: parameters.basketId, arg}).key
+
     return useQuery(
-        ['/baskets', parameters.basketId, arg],
+        queryKey,
```

Comment:
Now, there is this thing called `queryMap`. And it's being reused in both queries and mutations. ← inspired by the previous idea of having a query factory

Reply:
I think the `queryMap` idea is doing something similar to what we were doing in `getCacheUpdateMatrix`, right?
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/98b57269d2d8f9c7250d94dd5dedebfc3200f8b0/packages/commerce-sdk-react/src/hooks/ShopperBaskets/mutation.ts#L181

The `queryMap` concept maps query endpoints to its queryKey.

I think the suggested queryKey factory concept was a little bit different. It was a central location where all the query keys are grouped and defined. This central place with all the queryKeys and combinations of queryKeys then can be used in queries and mutations.
We decided to simplify and manually define the Query Key invalidation matrix manually in the first interactions.